napari/layers/labels/labels.py

```diff
@@ -1632,15 +1579,8 @@ def data_setitem(self, indices, value, refresh=True):
             # array, or a NumPy-array-backed Xarray, is the slice a view and
             # therefore updated automatically.
             # For other types, we update it manually here.
-            slice_input = self._slice.slice_input
-            point = np.round(
-                self.world_to_data(slice_input.world_slice.point)
-            ).astype(int)
-            pt_not_disp = {
-                dim: point[dim] for dim in slice_input.not_displayed
-            }
-            displayed_indices = index_in_slice(indices, pt_not_disp)
-            self._slice.image.raw[displayed_indices] = value
+            self._slice.image.raw[displayed_indices] = visible_values
+            # self._slice.image.view[displayed_indices] = self.colormap._map_to_gpu(value)
```

Comment:
Remove the comment? It might be worth adding a comment mentioning that the view must also be updated but is done below only if contour == 0 (why?)

Reply:
It is done only if `contour == 0` because in other situation we need to recalculate contours, so use `_raw_to_displayed method`